### ISSUE TITLE
Kamel 1.0 fails on install on mac

### ISSUE THREAD

**Issue by jasoncarreira** [apache/camel-k#1543]
I had a previous kamel CLI RC installed and just upgraded to 1.0. I installed initially with homebrew and have also uninstalled that way and installed from the mac binary on GitHub. Both give the same error:

kamel install
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1441556]

goroutine 1 [running]:
k8s.io/apimachinery/pkg/runtime.(*Scheme).convertToVersion(0xc0003bfdc0, 0x24b9a00, 0x29c7000, 0xc00040a460, 0x0, 0x0, 0x26c15e0, 0xc00040a460, 0x2a3fac0, 0x26c15e0)
	k8s.io/apimachinery@v0.18.2/pkg/runtime/scheme.go:527 +0x296
k8s.io/apimachinery/pkg/runtime.(*Scheme).UnsafeConvertToVersion(...)
	k8s.io/apimachinery@v0.18.2/pkg/runtime/scheme.go:493
k8s.io/apimachinery/pkg/runtime.unsafeObjectConvertor.ConvertToVersion(0xc0003bfdc0, 0x29c7000, 0xc00040a460, 0x0, 0x0, 0x0, 0x0, 0x29c7000, 0xc00040a460)
	k8s.io/apimachinery@v0.18.2/pkg/runtime/helper.go:39 +0x5f
k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).Decode(0xc00040a1e0, 0xc0000e0600, 0x15a, 0x600, 0xc0004e6780, 0x0, 0x0, 0x15a, 0x600, 0xc000359d00, ...)
	k8s.io/apimachinery@v0.18.2/pkg/runtime/serializer/versioning/versioning.go:171 +0x34c
k8s.io/client-go/rest.Result.Error(0xc0000e0600, 0x15a, 0x600, 0xc000359d00, 0x10, 0x29b0fc0, 0xc00040a320, 0x199, 0x3fc6eb0, 0xc00040a1e0, ...)
	k8s.io/client-go@v12.0.0+incompatible/rest/request.go:1218 +0x11a
github.com/apache/camel-k/pkg/install.installCRD(0x29fa480, 0xc0002c7c80, 0x2a4a9e0, 0xc000290b10, 0x2729c69, 0x13, 0x273a4a3, 0x1d, 0x0, 0x272c533, ...)
	github.com/apache/camel-k@/pkg/install/cluster.go:197 +0x42f
github.com/apache/camel-k/pkg/install.SetupClusterWideResourcesOrCollect(0x29fa480, 0xc0002c7c80, 0xc0000b79c8, 0x0, 0xc000500042, 0x5)
	github.com/apache/camel-k@/pkg/install/cluster.go:48 +0xb0
github.com/apache/camel-k/pkg/cmd.(*installCmdOptions).install(0xc000159b00, 0xc00003a840, 0x3810e30, 0x0, 0x0, 0x0, 0x0)
	github.com/apache/camel-k@/pkg/cmd/install.go:219 +0x1502
github.com/apache/camel-k/pkg/cmd.newCmdInstall.func1(0xc00003a840, 0x3810e30, 0x0, 0x0, 0x0, 0x0)
	github.com/apache/camel-k@/pkg/cmd/install.go:62 +0xa4
github.com/spf13/cobra.(*Command).execute(0xc00003a840, 0x3810e30, 0x0, 0x0, 0xc00003a840, 0x3810e30)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001a0dc0, 0xc0002c7c80, 0xc0000be020, 0x2)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	github.com/apache/camel-k@/cmd/kamel/main.go:47 +0x1af

**Comment by nicolaferraro** [comment]
@jasoncarreira I suspect this can be related to the type of cluster you're targeting rather than your local OS. What cluster are you using?

**Comment by nicolaferraro** [comment]
Do you hit the same error with `kamel version` or `kamel init Example.java`?

**Comment by jasoncarreira** [comment]
kamel version returns the version successfully. Kamel init hello.groovy doesn't seem to do anything, but doesn't through errors.

**Comment by jasoncarreira** [comment]
I'm using AKS on Azure. The 0.3.4 release would install on AKS but I couldn't get it to run. The previous 1.0-RC I had wouldn't install on AKS but didn't throw errors.

**Comment by pjcubero** [comment]
I'm trying reinstall Kamel-K over minishift, now I have the version 0.3.4, and I have the same error that from Jason Carreira... I don't know why...

**Comment by chenrui333** [comment]
running into the same issue with [kamel 1.3.0 build](https://github.com/Homebrew/homebrew-core/pull/67984)
```
$  /usr/local/Cellar/kamel/1.3.0/bin/kamel
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x24c0fe6]

goroutine 1 [running]:
github.com/spf13/cobra.(*Command).Name(0x0, 0xc00019a1b0, 0x0)
	github.com/spf13/cobra@v1.0.0/command.go:1258 +0x26
github.com/apache/camel-k/pkg/cmd.createKamelWithModelineCommand(0x2d78f60, 0xc0004b05c0, 0xc00019a1b0, 0x0, 0x0, 0x10, 0x4400108, 0xc000458be0, 0xc0b6d6ff9a, 0x3ac7b00, ...)
	github.com/apache/camel-k/pkg/cmd/modeline.go:92 +0xfc
github.com/apache/camel-k/pkg/cmd.NewKamelWithModelineCommand(0x2d78f60, 0xc0004b05c0, 0xc00019a1b0, 0x1, 0x1, 0xc0001c5f48, 0x1048aca, 0x3a43c20, 0xc000000180, 0xc0001c5f78, ...)
	github.com/apache/camel-k/pkg/cmd/modeline.go:64 +0xa5
main.main()
	command-line-arguments/main.go:45 +0x165
```

build log in here, https://github.com/Homebrew/homebrew-core/runs/1623638028

**Comment by chenrui333** [comment]
`kamel --version` and `kamel init Example.java` both work for me in my local.

**Comment by chenrui333** [comment]
Any update on this issue?

**Comment by chenrui333** [comment]
got resolved in [kamel 1.3.1](https://github.com/Homebrew/homebrew-core/pull/70404)

**Comment by apupier** [comment]
I hit very similar stacktrace than the initial one:

```
sh-5.0$ kamel install --olm=false
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8def36]

goroutine 1 [running]:
k8s.io/apimachinery/pkg/runtime.(*Scheme).convertToVersion(0xc0002056c0, 0x1b60900, 0x212c6c0, 0xc000442320, 0x0, 0x0, 0x1db2c20, 0xc000442320, 0x2198f00, 0x1db2c20)
        k8s.io/apimachinery@v0.18.9/pkg/runtime/scheme.go:488 +0x296
k8s.io/apimachinery/pkg/runtime.(*Scheme).UnsafeConvertToVersion(...)
        k8s.io/apimachinery@v0.18.9/pkg/runtime/scheme.go:454
k8s.io/apimachinery/pkg/runtime.unsafeObjectConvertor.ConvertToVersion(0xc0002056c0, 0x212c6c0, 0xc000442320, 0x0, 0x0, 0x0, 0x0, 0x212c6c0, 0xc000442320)
        k8s.io/apimachinery@v0.18.9/pkg/runtime/helper.go:39 +0x5f
k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).Decode(0xc000442140, 0xc000234c00, 0x1a6, 0x600, 0xc000668ae0, 0x0, 0x0, 0x1a6, 0x600, 0xc00030a010, ...)
        k8s.io/apimachinery@v0.18.9/pkg/runtime/serializer/versioning/versioning.go:171 +0x34c
k8s.io/client-go/rest.Result.Error(0xc000234c00, 0x1a6, 0x600, 0xc00030a010, 0x10, 0x21151c0, 0xc000442280, 0x193, 0x7f7a91b510a0, 0xc000442140, ...)
        k8s.io/client-go@v12.0.0+incompatible/rest/request.go:1269 +0x10d
github.com/apache/camel-k/pkg/install.installCRD(0x2161220, 0xc0006dc440, 0x21a4f80, 0xc00044d140, 0x1e29caf, 0x13, 0x1e128bb, 0x2, 0x1e3cc28, 0x1d, ...)
        github.com/apache/camel-k/pkg/install/cluster.go:218 +0x466
github.com/apache/camel-k/pkg/install.SetupClusterWideResourcesOrCollect(0x2161220, 0xc0006dc440, 0xc000929a08, 0x0, 0x2, 0xc0004f3968)
        github.com/apache/camel-k/pkg/install/cluster.go:49 +0xc8
github.com/apache/camel-k/pkg/cmd.(*installCmdOptions).install(0xc0006278c0, 0xc0006ee840, 0xc00041d700, 0x0, 0x1, 0x0, 0x0)
        github.com/apache/camel-k/pkg/cmd/install.go:232 +0x15e4
github.com/apache/camel-k/pkg/cmd.newCmdInstall.func1(0xc0006ee840, 0xc00041d700, 0x0, 0x1, 0x0, 0x0)
        github.com/apache/camel-k/pkg/cmd/install.go:62 +0xa4
github.com/spf13/cobra.(*Command).execute(0xc0006ee840, 0xc00041d6f0, 0x1, 0x1, 0xc0006ee840, 0xc00041d6f0)
        github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002b3080, 0xc0006dc400, 0xc00004e0c0, 0x3)
        github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        github.com/apache/camel-k/cmd/kamel/main.go:48 +0x17b
        
```

context:
- kamel 1.3.1
- in CodeReady Workspaces from Red Hat Developer sandbox
